packages/ses/src/tame-domains.js

```diff
@@ -0,0 +1,44 @@
+// @ts-check
+
+import {
+  TypeError,
+  globalThis,
+  getOwnPropertyDescriptor,
+  defineProperty,
+} from './commons.js';
+
+export function tameDomains(domainTaming = 'safe') {
+  if (domainTaming !== 'safe' && domainTaming !== 'unsafe') {
+    throw new TypeError(`unrecognized domainTaming ${domainTaming}`);
+  }
+
+  if (domainTaming === 'unsafe') {
+    return;
+  }
+
+  // Protect against the hazard presented by Node.js domains.
+  if (typeof globalThis.process === 'object' && globalThis.process !== null) {
+    // Check whether domains were initialized.
+    const domainDescriptor = getOwnPropertyDescriptor(
+      globalThis.process,
+      'domain',
+    );
+    if (domainDescriptor !== undefined && domainDescriptor.get !== undefined) {
```

Comment:
Why the `get` test?

Reply:
This is an important distinction and I’ve added a comment to the effect. Node.js initializes process with a process.domain === null and replaces this with a get/set pair when `domain` initializes.